### ISSUE TITLE
Does not play well with releases?

### ISSUE THREAD

**Issue by jaydoane** [herenowcoder/reprise#6]
I'm using https://github.com/bitwalker/exrm to generate releases, and have found what appears to be an incompatibility. While in most cases, one would probably want to generate a release for prod environment, the default case uses dev, and I'm seeing this error:
```
14:15:21.523 [error] GenServer Reprise terminating
Last message: :wake
State: {{{2014, 10, 30}, {14, 15, 20}}, 1000}
** (exit) an exception was raised:
    ** (Mix.Error) Cannot access build without an application name, please ensure you are in a directory with a mix.exs file and it defines an :app name under the project configuration
        (mix) lib/mix.ex:235: Mix.raise/1
        (mix) lib/mix/project.ex:303: Mix.Project.compile_path/1
        (mix) lib/mix/project.ex:343: Mix.Project.load_paths/1
        (reprise) lib/reprise.ex:127: Reprise.Runner.beams/0
        (reprise) lib/reprise.ex:135: Reprise.Runner.beam_modules/0
        (reprise) lib/reprise.ex:162: Reprise.Runner.go/2
        (reprise) lib/reprise.ex:81: Reprise.Server.handle_info/2
        (stdlib) gen_server.erl:599: :gen_server.handle_msg/5

```
So far, the workarounds I've come up with are to either force the release environment with ```MIX_ENV=prod mix release```, or to omit reprise entirely from the applications list, and start it manually.

It looks like reprise uses Mix to figure out exactly which beams to reload, so replacing it may not be trivial. Perhaps just surround line 127 with a try and return the empty list if Mix is unavailable? Any other ideas?

**Comment by herenowcoder** [comment]
Hi!

As I'm merging what @yrashk has posted wrt to releases, can you please remind me what was your need exactly?

This is how I understand it:
- there was an error during release generation, sure it's annoying, I guess that was the reason for you to patch it
- in general, does Reprise make sense inside a dev release? I guess it does

**Comment by jaydoane** [comment]
Yes, I agree that generating a dev release is unusual, but since it's also the default behavior, it would seem less surprising to not crash. In any case, a dev release could be used to test a release lifecycle, while still using dev configuration, which would seem useful, especially if we could also get the benefits of reprise in the shell :)

**Comment by herenowcoder** [comment]
I merged your remaining PR and pushed it as 0.4.0 to Hex.

If it'd be still dissatisfying, please reopen.

cheers!
